src/Microsoft.PowerShell.Commands.Utility/commands/utility/select-object.cs

```diff
@@ -414,7 +419,7 @@ private void ProcessParameter(MshParameter p, PSObject inputObject, List<PSNoteP
                 }
             }
 
-            if (expressionResults.Count == 0)
+            if (expressionResults.Count == 0 && !ex.HasWildCardCharacters)
             {
                 //Commented out for bug 1107600
```

Comment:
I generally see little comments in code and tests and thought it was best practice. I will make more comments.
I think @PowerShellTeam  needs to mark this moment in documentation: "make more comments in your code".